### DISCUSSION TITLE
Warning for wrong layout

### DISCUSSION THREAD

**Discussion by muddv** [monkeytypegame/monkeytype#4365] [Ideas]
**Problem**: If user has has several keyboard layouts for different languages, mistakes can happen if user forgot to switch language.

**Solution**: add a warning similar to the "Caps Lock" warning to prevent user from typing in the wrong language.

**Possible approach**: We can use [`Keyboard: getLayoutMap()`](https://developer.mozilla.org/en-US/docs/Web/API/Keyboard/getLayoutMap) method and research available polyfills for Firefox and Safari or use a different approach, while waiting for this method to be implemented by all major browsers. I can submit a pr with this feature if you think this is a good idea.

**Comment by Miodec** [comment]
Im pretty sure this would only work when the layout is set by the system. Many people use external software or even reprogram the keyboards themselves (with things like QMK) and that definitely not be picked up by this API.

**Comment by muddv** [comment]
True, I can test what this method picks up with QMK, but probably it will produce false warnings, but maybe this warning turned off entirely? Those who set language with system may still benefit from this.
Maybe with all the possible problems this should not be added after all
